### ISSUE TITLE
AddModel Doesn't rehydrate Persist

### ISSUE THREAD

**Issue by jchamb** [ctrlplusb/easy-peasy#444]
Have a project I'm working on where we use a lot of addModel as we lazy load in different parts of our app. Went to persist some filtering options on one such Model but it never gets rehydrated.

I didn't see anything in the documentation about manually calling it to rehydrate for a specific model. Is there anything I missing?

Thanks.

**Comment by ctrlplusb** [comment]
Hey @jchamb;

This is 100% a bug and unfortunately not a very straightforward one to resolve. I am trying to put aside some time to resolve it. Hopefully won't take too long. 💜

**Comment by ctrlplusb** [comment]
Hmm, ok, I have figured out how to get it to rebind with a small change. The rehydration experience would be ok for storage engines that are synchronous. However, if you are using an asynchronous store this may provide some jarring UX. This is because your dynamically added model would initially show the initial state values for your dynamically added model, but then would switch to the rehydrated state after it gets resolved.

One option would be to return a `Promise` from `addModel` which would get resolved after rehydration has completed. In the case of synchronous stores it would immediately resolve.

**Comment by ctrlplusb** [comment]
Ok, so I feel like the API I have come up with this feels safe enough to add for now.

Basically, instead of `undefined`, the `store.addModel` API will return an object, which will contain a `resolveRehydration` property. This would only be needed if you are using an asynchronous storage engine coupled with your dynamically added model.

Example usage:

```typescript
import React, { useState, useEffect } from 'react';
import { useStore } from 'easy-peasy';

function ProductsStoreLoader({ children }) {
  const store = useStore();
  const [storeLoaded, setStoreLoaded] = useState(false);
  useEffect(() => {
    const { resolveRehydration } = store.addModel('products', productsModel);
    resolveRehydration().then(() => setStoreLoaded(true);
    return () => {
      store.removeModel('products');
    };
  }, []);
  return storeLoaded ? React.Children.only(children) : null;
}
```

**Comment by ctrlplusb** [comment]
I have included this in the latest alpha release of #459 💜

**Comment by jchamb** [comment]
@ctrlplusb  sorry I missed this last week. This looks good to me and I think should take care of our use cases with some slight structure tweaking on our end. I'll give this a go first chance I get this week and let you know of any problems I stumble upon. 

Thanks for getting this squeezed in for 3.4 as well, this will def be huge fix for my team!
